codex/sales.nim

```diff
@@ -105,14 +105,14 @@ proc new*(
     subscriptions: @[],
   )
 
-proc remove(sales: Sales, agent: SalesAgent) {.async.} =
+proc remove(sales: Sales, agent: SalesAgent) {.async: (raises: [CancelledError]).} =
   await agent.stop()
   if sales.running:
     sales.agents.keepItIf(it != agent)
 
 proc cleanUp(
     sales: Sales, agent: SalesAgent, reprocessSlot: bool, returnedCollateral: ?UInt256
-) {.async.} =
+) {.async: (raises: [CancelledError]).} =
```

Comment:
I think we should avoid allowing cleanup routines to be cancelled, and thus should raise `[]` here. For each await in this routine, we can prefix with `noCancel`, eg `await noCancel sales.context.reservations.returnBytesToAvailability` which will prevent those routines from being cancelled and raising `CancelledError` (so that exception doesn't need to be handled).

Reply:
Yes, I agree. I'd suggest to solve this in #1234, which is based on this PR.